job-server/src/main/scala/spark/jobserver/util/ManagerLauncher.scala

```diff
@@ -42,9 +42,9 @@ class ManagerLauncher(systemConfig: Config, contextConfig: Config, masterAddress
       if (deployMode == "client") {
         val gcFilePath = new File(contextDir, gcFileName).toString()
         loggingOpts += s" -DLOG_DIR=$contextDir"
-        gcOPTS += s" -Xloggc:$gcFilePath"
+        gcOPTS = s"-Xloggc:$gcFilePath " + gcOPTS
       } else {
-        gcOPTS += s" -Xloggc:$gcFileName"
+        gcOPTS = s"-Xloggc:$gcFileName " + gcOPTS
```

Comment:
This should probably be broken off to its own small PR

Reply:
Here is the PR: https://github.com/spark-jobserver/spark-jobserver/pull/1292